aws-cpp-sdk-core/source/platform/android/FileSystem.cpp

```diff
@@ -158,6 +159,39 @@ Aws::String GetHomeDirectory()
     return Aws::Platform::GetCacheDirectory();
 }
 
+Aws::String Dirname(const Aws::String& pathname)
+{
+    if (pathname.length() == 0)
+    {
+        // Pathname is empty
+        return Aws::String(".");
+    }
+
+    Aws::String::const_reverse_iterator last_path_delim =
+                                        std::find(
+                                                pathname.rbegin(),
+                                                pathname.rend(),
+                                                Aws::FileSystem::PATH_DELIM);
+
+    if (last_path_delim == pathname.rend())
+    {
+        // No path delimiters found
+        return Aws::String(".");
+    }
+    else if ((last_path_delim.base() - pathname.begin()) == 1)
```

Comment:
check if `pathname.front() == PATH_DELIM` instead.
It's much easier to reason about the logic that way.